ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureStorageClient.java

```diff
@@ -0,0 +1,155 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud.azure;
+
+import com.azure.core.exception.UnexpectedLengthException;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.Context;
+import com.azure.core.util.FluxUtil;
+import com.azure.storage.blob.BlobServiceAsyncClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.batch.BlobBatch;
+import com.azure.storage.blob.batch.BlobBatchStorageException;
+import com.azure.storage.blob.models.AccessTier;
+import com.azure.storage.blob.models.BlobHttpHeaders;
+import com.azure.storage.blob.models.BlobProperties;
+import com.azure.storage.blob.models.BlobRange;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.azure.storage.blob.models.BlockBlobItem;
+import com.azure.storage.blob.models.DownloadRetryOptions;
+import com.azure.storage.blob.specialized.BlockBlobAsyncClient;
+import com.github.ambry.cloud.CloudBlobMetadata;
+import com.github.ambry.commons.BlobId;
+import com.github.ambry.commons.FutureUtils;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Abstract class to encapsulate ABS client operations.
+ */
+public interface AzureStorageClient {
+  /**
+   * Visible for testing.
+   * @return the underlying {@link BlobServiceClient}.
+   */
+  public BlobServiceAsyncClient getStorageClient();
+
+  /**
+   * Creates a new block blob, or updates the content of an existing block blob asynchronously.
+   * @param blobId {@link BlobId} of the blob to upload.
+   * @param data The data to write to the blob.
+   * @param length The exact length of the data. It is important that this value match precisely the length of the
+   * data provided in the {@link InputStream}.
+   * @param headers {@link BlobHttpHeaders}
+   * @param metadata Metadata to associate with the blob.
+   * @param tier {@link AccessTier} for the destination blob.
+   * @param contentMd5 An MD5 hash of the block content.
+   * @param requestConditions {@link BlobRequestConditions}
+   * @return a {@link CompletableFuture} of type {@link Void} that will eventually complete when block blob is uploaded
+   *         or will contain an exception if an error occurred.
+   */
+  public CompletableFuture<Void> uploadWithResponse(BlobId blobId, InputStream data, long length,
```

Comment:
nit: we don't need public modifiers for interface methods.

Reply:
Fixed